lib/timers.js

```diff
@@ -580,13 +580,15 @@ function processImmediate() {
   while (immediate) {
     domain = immediate.domain;
 
-    if (!immediate._onImmediate)
-      continue;
```

Comment:
Yep, that works (since we are moving onto the next immediate before `continue`ing rather than just continuing as in the original if branch, which is what led to the infinite loop).

I think I misunderstood your first comment above and tested the wrong thing.